### PR TITLE
Dependency updates for test-infra

### PR DESCRIPTION
**Pull request by cblecker**
- Swap github.com/ghodss/yaml for sigs.k8s.io/yaml
- Update k8s.io packages to kubernetes 1.12
- Update dep to 0.5.0

### CODE REVIEW

**Comment by cblecker** [comment]
@fejta @BenTheElder fought with this, but it's actually working now. was looking to run updates on peribolos in k/org, but saw there were a number of things that should be updated here first dep wise.

**Comment by BenTheElder** [comment]
FYI
/this-is-not-fine

File: Gopkg.toml [hunk 1]
```diff
@@ -106,3 +106,7 @@ required = [
   name = "github.com/jteeuwen/go-bindata"
   # latest, unlikely to change as go-bindata is no longer maintained
   revision = "6025e8de665b31fa74ab1a66f2cddd8c0abf887e"
+
+[[constraint]]
+  name = "gopkg.in/yaml.v2"
```
Comment (BenTheElder) [inline Gopkg.toml]: reason?

Comment (BenTheElder) [reply Gopkg.toml]: (let's leave a comment?)

**Comment by stevekuznetsov** [comment]
Nice!

File: Gopkg.toml (hunk 1 above)
Comment (cblecker) [reply Gopkg.toml]: Added reason (sigs.k8s.io/yaml needs a more updated version, but lacks that definition. I opened https://github.com/kubernetes-sigs/yaml/pull/11 to add that)

Comment (BenTheElder) [reply Gopkg.toml]: Thanks!

**Pull request merged**